yjit/src/backend/x86_64/mod.rs

```diff
@@ -338,7 +338,10 @@ impl Assembler
                     // Load each operand into the corresponding argument
                     // register.
                     for (idx, opnd) in opnds.into_iter().enumerate() {
-                        asm.load_into(C_ARG_OPNDS[idx], *opnd);
+                        // branch_stub_hit trampoline loads C_ARG_OPNDS, but we don't need anything for it.
+                        if C_ARG_OPNDS[idx] != *opnd {
+                            asm.load_into(C_ARG_OPNDS[idx], *opnd);
```

Comment:
Could you solve this in the backend instead? This could probably benefit the A64 target as well if `load_into` just became a no-op if the args were equal.

Reply:
I did that first and then switched back to this idea. We currently try to minimize the number of passes by doing everything in `x86_split` in order to keep the compilation fast (ref: https://github.com/ruby/ruby/pull/7471, https://github.com/ruby/ruby/pull/7487). However, as of `x86_split`, the insn does not exist yet because it's pushed right here.

Reply:
Ah okay, maybe by "backend" you meant the x86_emit (and arm64's equivalent) while I considered passes as part of the backend. I'll see if I can do that.

Reply:
Ahh yeah "backend" was totally unclear in the wording. I specifically meant the `asm.load_into` function.

Reply:
Oh, I didn't think of it, and that sounds nice :) will do that

Reply:
That worked nicely https://github.com/ruby/ruby/pull/7537/commits/a4e9555bd44f9d5c6d084de3fabefea8c42a7581. Thanks!